src/cli/onefuzz/debug.py

```diff
@@ -707,14 +716,23 @@ def __init__(self, name: str, creation_time: datetime):
             self.logger.info("Did not find any matching files to download")
             return None
 
-        if not all:
-            self.logger.info(f"Downloading only the {last} most recent files")
-            files = files[:last]
+        if granularity == "job":
+            self.logger.info(
+                f"Downloading all of the log files for each task associated with the job with id {job_id}"
+            )
+        elif granularity == "task":
```

Comment:
these are branches are no mutually exclusive. 
We can specify job, task and machine at the same time

Reply:
I think we should always download according to the most specific parameter provided. Otherwise, we could never download the log file for just a single machine.
Do you have something else in mind?

Reply:
oh i see. I missed the logic to set the granularity